utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,299 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    /// Increments the digits by 1. if the digits are empty, it will add
+    /// an element with value 1.
+    fn increment_abs_by_one(&mut self) -> Result<(), Error> {
+        for digit in self.digits.iter_mut().rev() {
+            *digit += 1;
+            if *digit < 10 {
+                #[cfg(debug_assertions)]
+                self.check_invariants();
+                return Ok(());
+            }
+
+            *digit = 0;
```

Comment:
Nit: I thought we changed this to `self.digits.pop()` or similar. Maybe you changed this because of the `.iter_mut()`. If this is so, please keep a counter variable for the number of trailing zeros and then truncate the digits array to that length before returning.

Reply:
Done.

Reply:
Suggestion: You don't need to include the `*digit = 0` because you always either truncate it or clear it.

Reply:
I kinda like having it for illustration. but you are right, I am going to remove it.